rfd/0088-passwordless-windows.md

```diff
@@ -0,0 +1,216 @@
+---
+authors: Tobiasz Heller (tobiasz.heller@goteleport.com)
+state: draft
+---
+
+# RFD 0088 - Passwordless for windows CLI
+
+## What
+
+Passwordless features for native windows CLIs, aka Windows Hello and FIDO
+support for CLI/`tsh`.
+
+This is a part of the [Passwordless RFD][passwordless rfd].
+
+Passwordless is available as a preview in Teleport 10.
+
+## Why
+
+Native, non-browser windows clients lack support for Windows Hello or FIDO
+devices. This RFD explores how we can achieve that support for `tsh` in a secure
+way.
+
+## Details
+
+Windows Hello and FIDO devices support is implemented via win32 API which is
+installed on Windows 10 19H1 or later.
+More details about API can be found in [API spec](https://github.com/microsoft/webauthn)
+or [documentation][webauthn API windows docs].
+
+```mermaid
+flowchart TB
+  WRP[WebAuthn RP]
+  subgraph CD[Client Device]
+    BC[Browser Client]
+    NC[Native Client]
+    subgraph WADLL[WEBAUTHN.dll]
+        PA[Platorm Authenticator]
+        CPA[Cross-platform Authenticator]
+    end
+  end
+  subgraph RA[Roaming Authenticator]
+    CTAP2[CTAP2 API]
+  end
+
+
+  WRP --> CD
+  BC --> WADLL
+  NC --> WADLL
+
+  CPA -. USB/NFC/BLE .-> CTAP2
+```
+
+`Webauthn.dll` is designed to work with Webauthn protocol, that's why
+integration with it results moslty in passing parameters from and back to
+server.
+
+<!--
+TODO(tobiaszheller): Check if RS256 requires any code changes on server side.
+-->
+Windows Hello works only with RS256 algorithm, while both FIDO2 and TouchID uses
```

Comment:
I am not sure if I understand. 
From what I understand, you can't specify key length. The only parameter that we need/can pass is that server supports RS256 - it will make windows-hello device available. 
I will remove that sentence from RFD, I am not sure if it brings anything to table.

Reply:
Do you know what algorithm / key length Windows uses for Hello keys?

Reply:
From what I was able to verify it's RSA2048 + SHA256 (both my local tests and some blog posts). I cannot find it though anyhow in windows docs :(

Reply:
That seems reasonable. I was just curious.